src/messaging/ExchangeContext.h

```diff
@@ -177,7 +183,7 @@ class DLL_EXPORT ExchangeContext : public ReliableMessageContext, public Referen
     void SetResponseTimeout(Timeout timeout);
 
 private:
-    Timeout mResponseTimeout; // Maximum time to wait for response (in milliseconds); 0 disables response timeout.
+    Timeout mResponseTimeout       = 0; // Maximum time to wait for response (in milliseconds); 0 disables response timeout.
```

Comment:
I was wondering if it is enough by just changing the initial value to 0,  for example:

1. A create an exchange
2. A send a message with flag  Messaging::SendMessageFlags::kExpectResponse and set timeout value --> response timer is activated
3. A receive response and called WillSendMessage
4. A send a message with flag  Messaging::SendMessageFlags::None  --> in this case, the response timer is still activated, right?

Reply:
The sequence of events here is as follows:

1. Create exchange.
2. Set timeout value (has to happen _before_ the send).
3. Send a message with `kExpectResponse` -> response timer is activated.
4. Receive response.  This cancels the response timer.  See the `CancelResponseTimer()` call in `ExchangeContext::HandleMessage`
5. Process response, call `WillSendMessage`.
6. Send message without `kExpectResponse`.  New response timer is not activated, because `kExpectResponse` is not set, exchange closes.

Reply:
nit: Shall we also reset mResponseTimeout to 0 in CancelResponseTimer()? otherwise the next message with flag Messaging::SendMessageFlags::kExpectResponse could use the previous timeout if it does not set the new timeout value.

Reply:
We could, but I thought that was the intent: that the user of an exchange will set up what it considers to be a "reasonable" timeout to use for all its messages.  That's certainly how CASE and PASE use it....